libraries/c_sdk/standard/mqtt/src/iot_mqtt_operation.c

```diff
@@ -716,8 +717,21 @@ void _IotMqtt_ProcessKeepAlive( IotTaskPool_t pTaskPool,
         {
             IotLogDebug( "(MQTT connection %p) PINGRESP was received.", pMqttConnection );
 
-            /* PINGRESP was received. Schedule the next PINGREQ transmission. */
+            /* PINGRESP was received. Schedule the next PINGREQ transmission.
+             * Control actual delay with another variable, but keep this here
+             * so a PINGREQ will be transmitted in the next execution. */
             pMqttConnection->nextKeepAliveMs = pMqttConnection->keepAliveMs;
+
+            /* Check if keep alive time is nonzero and greater than wait time. */
+            if( pMqttConnection->keepAliveMs > IOT_MQTT_RESPONSE_WAIT_MS )
```

Comment:
Only possibility of this condition not happening is when keepAliveMs is 0 , which is possible when awsMode is not selected. That means _Iot_ProceKeepAlive() should not be even scheduled ?

Reply:
That's correct, I'll just make this an assert